src/ClientOptionsTrait.php

```diff
@@ -115,6 +118,22 @@ private function buildClientOptions(array $options)
         // Keep track of the API Endpoint
         $apiEndpoint = $options['apiEndpoint'] ?? null;
 
+         // If the user provides a logger or the GOOGLE_SDK_DEBUG_LOGGING environment variable is set
+        // we attach a logger to the options array
+        // If we have the logger option SET to null, it should trump the environment variable AKA explicitely
+        // turning off logging
+        if (getenv('GOOGLE_SDK_DEBUG_LOGGING') ||
+        (array_key_exists('logger', $options) && $options['logger'] !== null)) {
+            // If we have the logger set to other than null and is not a LoggerInterface throw and exception.
+            if (isset($options['logger']) && !$options['logger'] instanceof LoggerInterface) {
+                throw new ValidationException(
+                    'the "Logger" option should be PSR-3 LoggerInterface compatible'
+                );
+            } elseif (getenv('GOOGLE_SDK_DEBUG_LOGGING')) {
+                $options['logger'] = new Logger();
+            }
```

Comment:
This logic would be a lot more readable like this

```php
if (!array_key_exists('logger', $options) && '1' === getenv('GOOGLE_SDK_DEBUG_LOGGING')) {
    $options['logger'] = new Logger();
}
// If the logger is supplied,  and is not a LoggerInterface throw and exception.
if (isset($options['logger']) && !$options['logger'] instanceof LoggerInterface) {
    throw new ValidationException(
        'the "logger" option should be PSR-3 LoggerInterface compatible'
    );
}
```

If you want to keep the `elseif` in there, that's fine too, since we don't _really_ need to perform the `instanceof` check for the default logger... although I don't think it's necessary, and I think this way is more readable.

Reply:
But what happens with you set the option `logger` to `nuil`. That would throw a `ValidationException`. This is still in the works right? But in the document I proposed that if you pass a `null` you are specifically disabling the logger for this client.

This is done for the following example in mind:
What if you have 3 different clients and two of them are having issues and you want to not log on one?
* You could enable the flag and pass the `null` option to the client you know is fine
* You have the flag disabled and pass loggers to the clients you want

Maybe we could instead of null pass a `false`? Or maybe another mechanism? As this is absolutely still in the design phase. We could have this discussion in the document :)

Reply:
> But what happens with you set the option logger to nuil. That would throw a ValidationException. This is still in the works right?

No, because in that case, `isset($options['logger'])` would return `false`



```

Reply:
You are correct. The behaviour I was thinking of `isset` is the one from` array_key_exists`.